src/nlr/Layer.h

```diff
@@ -53,8 +53,10 @@ class Layer
 
     void setLayerOwner( LayerOwner *layerOwner );
     void addSourceLayer( unsigned layerNumber, unsigned layerSize );
+    void removeSourceLayer( unsigned sourceLayer );
     const Map<unsigned, unsigned> &getSourceLayers() const;
-
+    const double *getWeightMatrix( unsigned sourceLayer ) const;
+    void reduceIndexFromAllMaps( unsigned startIndex );
```

Comment:
reduceIndexFromAllMaps is not an intuitive function, maybe add a comment about what it does